docs/docs/reference/project-files/glob-patterns.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Glob Patterns
+description: Support for importing data sources using glob patterns
+sidebar_label: Glob patterns
+sidebar_position: 10
+---
+
+Rill supports ingesting data from a group of files using glob patterns in the URI of source files. This allows you to specify a pattern that matches multiple files, making it easier to ingest data from a group of related files. Let's explore how glob patterns work in Rill.
+
+## Ingesting Data with Glob Patterns
+
+To ingest data using glob patterns, you include the pattern in the URI of the source files. Here's an example URI that will ingest all Parquet files in the `my-bucket` bucket that were created in January 2023:
+`
+gs://my-bucket/v=1/y=2023/m=1/*.parquet
+`
+
+By default, Rill applies certain limits when using glob patterns to ingest data. The default limits are as follows:
+- **Total size of all matching files**: 10GB
+- **Total file matches**: 1000
+- **Total files listed**: 1 million
+
+These limits can be configured in the `source.yaml` file. To modify the default limits, you can update the source.yaml file with following fields:
+ - `glob.max_total_size` : The maximum total size (in bytes) of all objects. 
+ - `glob.max_objects_matched`: The total file matches allowed.
+ - `glob.max_objects_listed`: The total files listed to match against the glob pattern. 
+
+For example, to increase the limit on the total bytes downloaded to 100GB, you would add the following line to the `source.yaml` file:
+`
+glob.max_total_size: 1073741824000
+`
+
+## Extract policies
+
+Rill also supports extracting a subset of data matching a glob pattern. There are two types of extract policies:
+1. **File based limits** : These limits restrict the number of files that are ingested.
+2. **Row based limits** : These limits restrict the amount of data that is ingested from each file.
+
+You can apply these policies individually or in combination to control the extraction process.
+  - Here are the possible combination and their semantics.
+    - If both `rows` and `files` are specified, each file matching the `files` clause will be extracted according to the `rows` clause.
+    - If only `rows` is specified, no limit on the number of files is applied. For example, getting a 1 GB `head` extract will download as many files as necessary.
+    - If only `files` is specified, each file upto limit will be fully ingested.
+
+Each policy can be configured by specifying two parameters:
+1. **Size** : The number of files/size of data to be fetched.
+2. **Strategy** : The strategy to fetch data. Currently, only **Head** (first n up to size) or **Tail** (last n up to size) is supported.
```

Comment:
1. Since they are keys, render as code **`size`**
2. Drop space before `:`
3. For `size`, clarify the cases – something like "the size of data in bytes (for `rows`) or number or files (for `files`) to fetch"